DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/DelegateApprovals/_unapprovedDelegateExpandable.cshtml

```diff
@@ -52,7 +52,11 @@
         Approve
       </button>
     </form>
-    <a class="nhsuk-button delete-button nhsuk-u-margin-bottom-0" href="" role="button">
+    <a class="nhsuk-button delete-button nhsuk-u-margin-bottom-0"
+       asp-controller="DelegateApprovals"
+       asp-action="DelegateRejectionPage"
+       asp-route-delegateId="@Model.Id"
+       role="button">

```

Comment:
Came here because I stumbled upon this being an `<a>` while testing something else. Is there a reason this can't be a button? If we're using an `<a>` as a button it's not enough to just add `role="button"` to it, in particular it will still be missing keyboard controls e.g. spacebar to submit. See the [MDN docs on the button role here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role).
We can replace that functionality using js but that's extra complexity and breaks if someone has js disabled; ideally we would just always use a button.

Reply:
Ah, great question! 

If we used a button to navigate the site, it would have to either have an onclick event of navigating the site using JS (setting location href, but this approach is JS dependent) or a form that needs to be submitted (which is clunky and not following MVC standard way of navigating the site). If the links fail AA accessibility standards we should either:
* Fix it with more spec than role="button" (note that role="button" seems to be used on Register to vote which is either AA or AAA compliant - can't remember exactly).
* Use links instead, which is something that HEE didn't want in terms of the design choice.

Reply:
RtV is AAA rated I believe. Surprising that they can get away with that (but I agree they definitely do this!). I'm struggling to find the actual requirement to see whether it falls under AA or AAA or whatever but in all the w3c docs they say you must add support for enter and spacebar, and [here](https://www.w3.org/TR/2016/WD-wai-aria-practices-1.1-20160317/examples/button/button.html) they specifically call an `<a>` without keypress handlers "non-compliant".

Reply:
Also I'm happy to concede that I don't understand enough of the context here but since the approve button is using a form I don't see why we can't make it a multi-button form with an approve action and reject action.

Reply:
Ah - I think we can change the endpoint to POST & use the form submit for consistency with Reject + compliance with the rules you found, but I don't think we'll be changing e.g. Log in and Register buttons into forms, if they're not truly forms and just for navigation 👍 :)